src/partition/repart.c

```diff
@@ -6239,6 +6241,55 @@ static int determine_auto_size(Context *c) {
         return 0;
 }
 
+static int make_staging_directory(Context *context) {
+        _cleanup_free_ char *p = NULL;
+        _cleanup_close_ int fd = -1;
+        int r;
+
+        assert(context);
+
+        r = tempfn_random(arg_node, "repart", &p);
```

Comment:
hmm, if arg_node refers to a physical block device, then this will create the temporary dir in /dev/?